flight/mcu/lib/teensyShared/rocketFSM.cpp

```diff
@@ -30,32 +30,38 @@
 #include "pins.h"
 #include "thresholds.h"
 
-rocketFSM::rocketFSM(pointers *ptr) { pointer_struct = ptr; }
+rocketFSM::rocketFSM(pointers *ptr) {
+    pointer_struct = ptr;
+    // get the linear accelration from the lowgimu
+    float linear_acceleration = pointer_struct->sensorDataPointer->highG_data.hg_az;
+    // storing pointer_struct variables locally for use
+    FSM_State rocket_state = pointer_struct->sensorDataPointer->rocketState_data.rocketState;
+    uint32_t launch_time = pointer_struct->rocketTimers.launch_time;
+    uint32_t burn_timer = pointer_struct->rocketTimers.burn_timer;
+    uint32_t burnout_time = pointer_struct->rocketTimers.burnout_time;
+    uint32_t coast_timer = pointer_struct->rocketTimers.coast_timer;
```

Comment:
These are all local variables, did you mean to assign to the member variables?

Reply:
> I like changing everything to member variables. All code outside the fsm class still refers to the values in the pointer struct though so that code will have to be updated.

@NPIPHI Everything that deals with the FSM is within the class, right?

Reply:
If we change the member variables to pointers to the pointer_struct instead of just values, doesn't that take care of our problem?

Reply:
To the best of my knowledge it would, and it'd also be a great place to introduce references to the code since the actual pointer will never have to change.

Reply:
I would say change the timers to pointers. the time (as opposed to timer) variables I think are constant so those don't necessarily need to be pointers. Also need pointers to whatever sensors we need, it looks like just accelerometers though.

Reply:
I think thresholds are better left in the thresholds.h file, but the timers should definitely be stored as pointers

Reply:
^